schemas/dab.draft.schema.json

```diff
@@ -1,16 +1,20 @@
 {
   "$schema": "https://json-schema.org/draft-07/schema",
-  "$id": "#/additionalProperties/version",
+  "$id": "https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json",
   "title": "Data API builder",
   "description": "Schema for Data API builder engine",
-  "additionalProperties": {
-    "version": "https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"
-  },
+  "additionalProperties": false,
   "type": "object",
   "properties": {
+    "$schema": {
```

Comment:
This doesn't seem to be valid usage of the `$schema` keyword. Per spec the usage is to be string uri . I recommend dropping the dollar sign and use `schemaUrl`
https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-01#name-the-schema-keyword

> 8.1.1. The "$schema" Keyword
> The "$schema" keyword is both used as a JSON Schema dialect identifier and as the identifier of a resource which is itself a JSON Schema, which describes the set of valid schemas written for this particular dialect.
>
> The value of this keyword MUST be a URI [RFC3986] (containing a scheme) and this URI MUST be normalized. The current schema MUST be valid against the meta-schema identified by this URI.

> If this URI identifies a retrievable resource, that resource SHOULD be of media type "application/schema+json".
>
> The "$schema" keyword SHOULD be used in the document root schema object, and MAY be used in the root schema objects of embedded schema resources. It MUST NOT appear in non-resource root schema objects. If absent from the document root schema, the resulting behavior is implementation-defined.
>
> Values for this property are defined elsewhere in this and other documents, and by other parties.

Reply:
Additionally
- document this new schema property in the pr description.
- the value of this property should then be used to find a Dab schema file whose root `$id` property matches the value set in this property.

Reply:
> This doesn't seem to be valid usage of the `$schema` keyword. Per spec the usage is to be string uri . I recommend dropping the dollar sign and use `schemaUrl` https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-01#name-the-schema-keyword
> 
> > 8.1.1. The "$schema" Keyword
> > The "$schema" keyword is both used as a JSON Schema dialect identifier and as the identifier of a resource which is itself a JSON Schema, which describes the set of valid schemas written for this particular dialect.
> > The value of this keyword MUST be a URI [RFC3986] (containing a scheme) and this URI MUST be normalized. The current schema MUST be valid against the meta-schema identified by this URI.
> 
> > If this URI identifies a retrievable resource, that resource SHOULD be of media type "application/schema+json".
> > The "$schema" keyword SHOULD be used in the document root schema object, and MAY be used in the root schema objects of embedded schema resources. It MUST NOT appear in non-resource root schema objects. If absent from the document root schema, the resulting behavior is implementation-defined.
> > Values for this property are defined elsewhere in this and other documents, and by other parties.

I think that this use of the keyword is supposed to be reserved for the root level of the document. Since we are describing how the `$schema` property is being used here as a part of the "properties" field at root level, using "$schema" here seems fine and the clearest way to describe how it is being used in the referenced JSON.

Reply:
"properties" field needs to contain all the properties like they should be present in the config, since we are not allowing any additional properties we need to define it, including all the root properties.